packages/replayio/src/utils/cachedFetch.ts

```diff
@@ -28,6 +29,17 @@ export async function cachedFetch(
         statusText: resp.statusText,
       });
     } else if (attempt < maxAttempts) {
+      if (shouldRetry) {
+        if (!shouldRetry(resp)) {
```

Comment:
This is a sync call and `resp` isn't that perfect for sync usage since to check API-returned errors u usually have to convert those using `await resp.text()` or `await resp.json()`. It would be great to give a resolved value to the callback here but I'm not sure how to decide what to give it, `.text()` seems to be best common denominator but it's going to be annoying to convert it to JSON within `shouldRetry`'s implementation.

Reply:
Maybe we don't need this functionality right now? When the need arises it won't be hard to implement it but at least the implementer will have a better context for what they'll have to do.

Reply:
That's a good point.